tests/unit/s2n_hkdf_test.c

```diff
@@ -399,17 +400,12 @@ static struct hkdf_test_vector tests[] = {
 
 int main(int argc, char **argv)
 {
-    struct s2n_hmac_state hmac;
+    struct s2n_hmac_state hmac = { 0 };
 
-    uint8_t prk_pad[MAX_PSEUDO_RAND_KEY_SIZE];
-    struct s2n_blob prk_result = { 0 };
-    EXPECT_SUCCESS(s2n_blob_init(&prk_result, prk_pad, sizeof(prk_pad)));
+    uint8_t prk_pad[MAX_PSEUDO_RAND_KEY_SIZE] = { 0 };
+    uint8_t output_pad[MAX_OUTPUT_SIZE] = { 0 };
 
-    uint8_t output_pad[MAX_OUTPUT_SIZE];
-    struct s2n_blob out_result = { 0 };
-    EXPECT_SUCCESS(s2n_blob_init(&out_result, output_pad, sizeof(output_pad)));
-
-    struct s2n_blob in_key_blob, salt_blob, info_blob, actual_prk_blob, actual_output_blob;
+    struct s2n_blob in_key_blob, salt_blob, info_blob, actual_prk_blob, actual_output_blob, prk_result, out_result = { 0 };
```

Comment:
You're actually only zero-initing `out_result` here. If you want to init every blob, every entry in this list needs `= { 0 }`